[gn] Use exec_script_allowlist in //llvm/utils/gn/.gn

dpranke opened the pull request:
Use exec_script_allowlist in //llvm/utils/gn/.gn

GN is moving to use exec_script_allowlist instead of exec_script_whitelist everywhere.

See crbug.com/389986807

dpranke committed:
Update .gn

Use exec_script_allowlist in //llvm/utils/gn/.gn

GN is moving to use exec_script_allowlist instead of exec_script_whitelist everywhere.

See crbug.com/389986807

zmodem commented:
I see the new spelling was added in January (https://gn.googlesource.com/gn/+/ed1abc107815210dc66ec439542bee2f6cbabc00), so this requires a fairly recent version of gn. I don't know if that's a problem for people or not.

Anyway, @nico knows the gn build best.

nico reviewed (approved):
lg, but please give me a bit to update the bots.

Do you want to update the comment right above too?

Could you also update https://github.com/llvm/llvm-project/blob/783201b184572a07efe2dc6b6b9110873421cf11/llvm/utils/gn/build/toolchain/BUILD.gn#L86 ? (Move it to top of file; update comment.)

Another idea could be to assign to one or the other depending on `gn_version`.

But I think fairly few people use this build, and it's fairly easy to update GN (just run `llvm/utils/gn/get.py`), so I think it's fine to just require a new(ish) version of GN.

dpranke committed:
Update BUILD.gn

Remove version check from utils/gn/build/toolchain/BUILD.gn

dpranke committed:
Add gn_version check in dotfile

Add a gn_version check in the llvm/util/gn/.gn dotfile, to
ensure we have a new enough version of GN.

nico reviewed (approved):
Thanks, looks great! I'll land this some time today, after making sure my bots have a new-enough GN.

nico commented:
> Do you want to update the comment right above too?

(Looks like this wasn't done yet.)

dpranke committed:
Update .gn

whoops, missed an "allowing" reference.

dpranke commented:
> > Do you want to update the comment right above too?
> 
> (Looks like this wasn't done yet.)

Whoops. Fixed.

dpranke committed:
remove change from dotfile; gn_version isn't defined there.

dpranke committed:
Move gn_version check to secondary/BUILD.gn

nico commented:
Hm, looks like the assert runs too late. On a machine where I hadn't bumped GN yet, I'm getting

```
ERROR at //.gn:9:25: Assignment had no effect.
exec_script_allowlist = []
                        ^
You set the variable "exec_script_allowlist" here and it was unused before it went
out of scope.
```

Ah well. If anyone runs into this, running `llvm/utils/gn/get.py` makes it go away.

dpranke commented:
Yeah, the dotfile is processed before the other files, so I'm not surprised.
